### PR TITLE
Backport Table Export Changes

### PR DESCRIPTION
**Pull request by kingthorin**
Backport the changes implemented in the newer generic component.
https://github.com/zaproxy/zaproxy/pull/3545#issuecomment-302499767
https://github.com/zaproxy/zaproxy/pull/3543
https://github.com/zaproxy/zaproxy/pull/3540
The generic component won't be available to extension until after the next core update.

Fixes zaproxy/zaproxy#3557

### CODE REVIEW

File: src/org/zaproxy/zap/extension/fuzz/httpfuzzer/ui/HttpFuzzResultsContentPanel.java [hunk 1]
```diff
@@ -152,11 +154,16 @@ public void actionPerformed(ActionEvent e) {
...
-                    JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(), Constant.messages.getString("fuzz.httpfuzzer.results.toolbar.button.export.showMessageSuccessful"));
                 } catch(Exception ex) {
+                    success = false;
                     JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(), Constant.messages.getString("fuzz.httpfuzzer.results.toolbar.button.export.showMessageError") + "\n" + ex.getLocalizedMessage());
                     logger.error("Export Failed: " + ex);
                 }
+             // Delay the presentation of success message, to ensure all the data was already flushed.
+                if (success) {
+                    JOptionPane.showMessageDialog(View.getSingleton().getMainFrame(),
+                            Constant.messages.getString("export.button.success"));
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/fuzz/httpfuzzer/ui/HttpFuzzResultsContentPanel.java]: It should use the old key.

**Comment by thc202** [comment]
The extension check could also be backported.

File: src/org/zaproxy/zap/extension/fuzz/httpfuzzer/ui/HttpFuzzResultsContentPanel.java [hunk 2]
```diff
@@ -136,13 +139,27 @@ public void itemStateChanged(ItemEvent e) {
 
         @Override
         public void actionPerformed(ActionEvent e) {
-            WritableFileChooser chooser = new WritableFileChooser();
+			WritableFileChooser chooser = new WritableFileChooser(Model.getSingleton().getOptionsParam().getUserDirectory()) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/fuzz/httpfuzzer/ui/HttpFuzzResultsContentPanel.java]: These statements need some indentation tweaks.

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/fuzz/httpfuzzer/ui/HttpFuzzResultsContentPanel.java]: Fixed

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
ping @psiinon 😀

**Comment by psiinon** [comment]
@kingthorin :)

**Pull request merged**